### ISSUE TITLE
aws/route53: Input for record.*.name != record.*.name on the output

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/terraform#1126]
This is probably low-prio + easy to do, but it's more important to discuss it here.

Look at the following code and tell me what would you expect the output to be?

```ruby
resource "aws_route53_zone" "primary" {
   name = "example.radeksimko.com"
}

resource "aws_route53_record" "sample" {
  zone_id = "${aws_route53_zone.primary.id}"
  name = "sample"
  type = "CNAME"
  ttl = "60"
  records = ["www.terraform.io"]
}

output "name" {
  value = "${aws_route53_record.sample.name}"
}
```

I would definitely expect `${aws_route53_record.sample.name}` to be `sample`, not the full domain (`sample.example.radeksimko.com`).
User will most likely want to use the full domain elsewhere instead of concatenating it (so it should really stay in the output), but I'm just not sure about the `name` here.

How about something like `aws_route53_record.sample.full_name` or `aws_route53_record.sample.domain` or `aws_route53_record.sample.fqdn`?

**Comment by catsby** [comment]
This is partially the result of https://github.com/hashicorp/terraform/pull/969 , where we allow people to not have to use the fully qualified `name`. Here, `name` represents "DNS domain name" 

- http://docs.aws.amazon.com/Route53/latest/APIReference/API_ChangeResourceRecordSets_Requests.html#API_ChangeResourceRecordSets_RequestBasicSyntax

Prior to #969 , you would have needed to specify the fully qualified name, so you would then also get `sample.example.radeksimko.com`. At least then, it wouldn't be surprise... 

In theory, we could add a new `computed` schema element that is ignored in configuration, and is set from the `name` value here? Does that sound sufficient?

**Comment by radeksimko** [comment]
> This is partially the result of #969 , where we allow people to not have to use the fully qualified name. 

yeah, except it's effectively now useless as [it triggers the update all the time](https://github.com/hashicorp/terraform/issues/1122)... :wink: 

I do have to use the full FQDN everywhere anyway if I want consistent behaviour and not recreate the record each time I run `terraform apply` :)

> In theory, we could add a new computed schema element that is ignored in configuration, and is set from the name value here? Does that sound sufficient?

That's basically what I meant and what I was suggesting. :+1: 

The whole confusion just comes from the fact that `input.name` != `output.name` - the same attribute on input may not always equal the attribute on output.

**Comment by radeksimko** [comment]
This has been fixed in #1279
